core/src/main/java/org/apache/carbondata/core/metadata/schema/table/DiskBasedDMSchemaStorageProvider.java

```diff
@@ -103,7 +103,7 @@ public DiskBasedDMSchemaStorageProvider(String storePath) {
   }
 
   @Override public List<DataMapSchema> retrieveSchemas(CarbonTable carbonTable) throws IOException {
-    checkAndReloadDataMapSchemas();
+    checkAndReloadDataMapSchemas(false);
```

Comment:
In case carbon table is non transactional don't touchFile

Reply:
retrieveSchemas no need touch md file

Reply:
I think you need to decide the true of false based on the `carbonTable.isTransactionalTable`

Reply:
I think,  it is not required to touch the md file in case of retrieve